### ISSUE TITLE
Field AuthToken in table usersessions for sessions without usernames equal to encoded sessionID

### ISSUE THREAD

**Issue by rwaaren** [domoticz/domoticz#3156]
Somewhere between V4.10366 and V4,10531, the generation of AuthToken| for sessionID's without usernames changed. Until 4.10366 the AuthToken was unique, after that it was equal to the encoded sessionID.
could it be the change to webserver/cWebem.cpp on feb 3rd ?

It seems that this can lead to errors in the log like

_2019-04-22 14:36:00.978  Error: SQL Query("INSERT INTO UserSessions (SessionID, Username, AuthToken, ExpirationDate, RemoteHost) VALUES ('b15a987a6c3ec0c3b64120993996776b', '', 'b15a987a6c3ec0c3b64120993996776b', '2019-04-22 14:46:00', '192.168.192.1')") : UNIQUE constraint failed: UserSessions.AuthToken_

end jan 2019 (V4.10366) 
```
138fd0ea9c025c3ee32ea9750e402ff5||a9f931d3e30f0a36ba67e525e89b7757|2019-01-31 04:46:00|192.168.192.1|2019-01-31 04:36:00
0dcab84cea9ef5e961336fa216bd0904||215cefa7f0fe2015015849053e687fae|2019-01-31 04:49:01|192.168.192.59|2019-01-31 04:39:01
bac90fcf8f3d781f94258518d6951d5a||831ee9c8edfad93b9687ce7f6d257aec|2019-01-31 04:49:01|192.168.192.220|2019-01-31 04:39:01
7641440b176257a86ba0e62677a98194||db7755f3ba9a80c09b04804026345fca|2019-01-31 04:49:01|192.168.192.97|2019-01-31 04:39:01
e7450388c277e8718e4bab34be6da747||22052b093c9d41fb5f74155e4e27e244|2019-01-31 04:58:00|192.168.192.1|2019-01-31 04:48:00
fb0c81b0ff395cb9722482984ef9701c||9ba628caaf2aa36905f20d4e1de84a5b|2019-01-31 05:02:01|192.168.192.220|2019-01-31 04:52:01
674a6abf4f9c03195ec5c252d3203519||6a49ee8edf36a6c2551745da3e40d3f6|2019-01-31 05:02:02|192.168.192.97|2019-01-31 04:52:02
3a52db5a446b4db869b111f27277207b||1b5a8807d5aac38a27bc39404f9a2985|2019-01-31 05:02:22|192.168.192.59|2019-01-31 04:52:22
```

end feb 2019 (V4.10531:

```
a0772b15751ee6984971ebbad5a787bb||a0772b15751ee6984971ebbad5a787bb|2019-02-28 04:58:00|192.168.192.1|2019-02-28 04:48:00
4805fe269587247b048d3523a676ad6e||4805fe269587247b048d3523a676ad6e|2019-02-28 05:02:01|192.168.192.243|2019-02-28 04:52:01
e4135ce6b706a35f3e377e1f06050391||e4135ce6b706a35f3e377e1f06050391|2019-02-28 05:02:08|192.168.192.59|2019-02-28 04:52:08
```

**Comment by rwaaren** [comment]
Investigated a bit more and it looks like that if two sessions in the same second try to access domoticz and both are without username, the generatUUID function generates the same AuthToken.
**table usersessions**
```
SessionID|Username|AuthToken|ExpirationDate|RemoteHost|LastUpdate
cc1dbe23e5e17f218be1c468a9b5f5e6||21a8b35c1f1401435f703c5e05805a1a|2019-04-23 22:58:00|127.0.0.1|2019-04-23 22:48:00
```

**logmessage**
`2019-04-23 22:48:00.906  Error: SQL Query("INSERT INTO UserSessions (SessionID, Username, AuthToken, ExpirationDate, RemoteHost) VALUES ('21a8b35c1f1401435f703c5e05805a1a', '', '21a8b35c1f1401435f703c5e05805a1a', '2019-04-23 22:58:00', '192.168.192.1')") : UNIQUE constraint failed: UserSessions.AuthToken`

**Comment by rwaaren** [comment]
Solved by PR #3158
